### PR TITLE
scheduler: Add option to change voting power distribution

### PR DESCRIPTION
**Pull request by abukosek**
This PR adds the ability to change the voting power distribution via consensus governance.
The two options currently available are linear (the original, where voting power is linearly proportional to the stake) and sqrt (new, where voting power is proportional to the square root of the stake).

TODO:
- [x] tests

### CODE REVIEW

**Review by kostko — COMMENTED** [review]
Can we update the governance txsource workload (in `cmd/debug/txsource/workload/governance.go`) to randomly change the voting power distribution via a parameter change proposal from time to time? So we can test that this actually works in E2E tests.

**Comment by peternose** [comment]
How come we decided to use `sqrt` and not `log`? Were any studies made?

**Comment by abukosek** [comment]
> How come we decided to use `sqrt` and not `log`? Were any studies made?

Don't know, but looking at Go's bigint implementation, it has `Sqrt` (https://pkg.go.dev/math/big#Int.Sqrt), but not `Log`, so it's a moot point :P (unless you guys want me to do polynomial approximation)

**Comment by peternose** [comment]
> Don't know, but looking at Go's bigint implementation, it has `Sqrt` (https://pkg.go.dev/math/big#Int.Sqrt), but not `Log`, so it's a moot point :P 

Agree. I have no idea which would be better. Intuition says that `Log` is the way to go (to lower the tops), but maybe falls to quickly. Keep `Sqrt`.

> (unless you guys want me to do polynomial approximation)

Please no :smile:

File: go/scheduler/api/api.go [hunk 1]
```diff
@@ -204,13 +204,25 @@ func (c *Committee) EncodedMembersHash() hash.Hash {
...
 // VotingPowerFromStake computes that by dividing by BaseUnitsPerVotingPower.
 //
 // NOTE: It's not that we're implementation-hiding the conversion though.
 // It's just that otherwise if we accidentally skip the `IsInt64`, it would
 // still appear to work, and that would be a bad thing to have in a routine
 // that's written multiple times.
-func VotingPowerFromStake(t *quantity.Quantity) (int64, error) {
+func VotingPowerFromStake(t *quantity.Quantity, distribution VotingPowerDistribution) (int64, error) {
 	powerQ := t.Clone()
 	if err := powerQ.Quo(&BaseUnitsPerVotingPower); err != nil {
```
[the remark this answers is not in the document]
Comment (abukosek) [reply go/scheduler/api/api.go:227]: I don't think so?
@kostko ?

[the remark this answers is not in the document]
Comment (kostko) [reply go/scheduler/api/api.go:228]: The reason for this scaling is to ensure the voting power doesn't overflow as it is limited to 64 bits. It is true that with a non-linear distribution (e.g. `sqrt`) such scaling may not be needed as `sqrt(10**19)` (max total supply in base units) is well under the limit.

[the remark this answers is not in the document]
Comment (peternose) [reply go/scheduler/api/api.go:228]: The explanation above should be inserted as a comment before the division. As things are now clear, I think we can optimize and remove unnecessary division in `sqrt` distribution.

[the remark this answers is not in the document]
Comment (abukosek) [reply go/scheduler/api/api.go]: Done :)

**Review by kostko — APPROVED** [review]

**Pull request merged**